index.js

```diff
@@ -17,4 +13,18 @@ module.exports = obj => {
       return { [entry[0]]: entry[1] };
     })
     .reduce((acc, curr) => Object.assign(acc, curr), {});
+}
+
+module.exports = obj => {
+  const promisifiedObject = promisifyObject(obj);
+
+  if (typeof obj === 'function') {
+    const promisiedFunction = promisify(obj);
+
+    Object.assign(promisiedFunction, promisifiedObject);
```

Comment:
This seems fine to me, but it's probably be better to check the type and only do `promisifyObject` on `obj` in the non-function case. (`s/promisied/promisified/g` tho)